packages/wallet/src/SingleAddressWallet.ts

```diff
@@ -213,23 +217,31 @@ export class SingleAddressWallet implements ObservableWallet {
       syncStatus: this.syncStatus
     });
     const tipBlockHeight$ = distinctBlock(this.tip$);
-    this.timeSettings$ = new PersistentDocumentTrackerSubject(
-      // TODO: it only really needs to be fetched once per epoch,
-      // so we should replace the trigger from tipBlockHeight$ to epoch$.
-      // This is a little complicated since there is a circular dependency.
-      // Some logic is needed to initiate a fetch if epoch is not available in store already.
+
+    this.#timeSettingsTrigger = new BehaviorSubject<void>(undefined);
+    this.#timeSettingsManualTracker$ = new PersistentDocumentTrackerSubject(
```

Comment:
Would this work?

```suggestion
    this.timeSettings$ = new PersistentDocumentTrackerSubject(
```

Reply:
I don't think it will work in this case: timeSettings$ is subscribed to, but epoch$ is not subscribed. In this case, timeSettings$ will not be refetched

Reply:
Updates are in [33d730e](https://github.com/input-output-hk/cardano-js-sdk/pull/338/commits/33d730e6c21ad834e9feec984870b90b526b2cb8)